version: '2.1'
services:
  grafana:
    container_name: grafana
    image: raymondmm/grafana
    ports:
      - 3000:3000
    volumes:
      - /var/docker/config/grafana:/etc/grafana:rw
      - /var/docker/data/grafana:/var/lib/grafana:rw
    restart: on-failure
    depends_on:
      influxdb:
        condition: service_healthy

  influxdb:
    container_name: influxdb
    image: influxdb
    ports:
      - 8086:8086
    volumes:
      - /var/docker/config/influxdb/influxdb.conf:/etc/influxdb/influxdb.conf:ro
      - /var/docker/data/influxdb:/var/lib/influxdb
    restart: on-failure
    healthcheck:
      test: ["CMD", "curl", "-sI", "http://127.0.0.1:8086/ping"]
      interval: 30s
      timeout: 1s
      retries: 24
      
  homeassistant:
    container_name: homeassistant
    image: homeassistant/raspberrypi2-homeassistant
    network_mode: "host"
    volumes:
      - /var/docker/config/homeassistant:/config
      - /etc/localtime:/etc/localtime:ro
      - /etc/letsencrypt:/etc/letsencrypt:ro
    #devices:
    #  - /dev/ttyUSB0:/dev/ttyUSB0:rwm
    #  - /dev/ttyUSB1:/dev/ttyUSB1:rwm
    restart: on-failure
    depends_on:
      influxdb:
        condition: service_healthy
      mosquitto:
        condition: service_started
    healthcheck:
      test: ["CMD", "curl", "-f", "http://127.0.0.1:8123"]
      interval: 30s
      timeout: 10s
      retries: 6

  #node-red:
  #  container_name: node-red
  #  image: nodered/node-red-docker:rpi-v8
  #  ports:
  #    - 1880:1880
  #  volumes:
  #    - /opt/node-red:/data
  #    - /etc/localtime:/etc/localtime:ro
  #    - /etc/timezone:/etc/timezone:ro
  #  restart: "no" # because the exit code(s) are wrong
  #  depends_on:
  #    mosquitto:
  #      condition: service_started
  #    #mqtt-bridge:
  #    #  condition: service_started
  #    homeassistant:
  #      condition: service_healthy
  #  healthcheck:
  #    test: ["CMD", "curl", "-f", "http://127.0.0.1:1880"]
  #    interval: 30s
  #    timeout: 10s
  #    retries: 5

  mosquitto:
    container_name: mosquitto
    image: robotany/mosquitto-rpi
    user: "1000:1000"
    ports:
      - 1883:1883
    volumes:
      - /etc/localtime:/etc/localtime:ro
      - /var/docker/config/mosquitto:/mosquitto/config:ro
      - /var/docker/data/mosquitto:/mosquitto/data
    restart: on-failure

  #mqtt-bridge:
  #  container_name: mqtt-bridge
  #  build:
  #    context: https://github.com/stjohnjohnson/smartthings-mqtt-bridge.git
  #    dockerfile: Dockerfile-rpi
  #  ports:
  #    - 8080:8080
  #  volumes:
  #    - /opt/mqtt-bridge:/config
  #  restart: on-failure
  #  depends_on:
  #    homeassistant:
  #      condition: service_healthy

  portainer:
    container_name: portainer
    image: portainer/portainer
    ports:
      - 9000:9000
    volumes:
      - /var/docker/data/portainer:/data:rw
      - /var/run/docker.sock:/var/run/docker.sock
    restart: "no" # because the exit code(s) are wrong
    networks:
      - default

  #organizr:
  #  container_name: organizr
  #  image: lsioarmhf/organizr
  #  ports:
  #    - 80:80
  #    - 443:443
  #  volumes:
  #    - /var/docker/config/organizr:/config:rw
  #    - /etc/letsencrypt:/etc/letsencrypt:ro
  #  environment:
  #    - PGID=1000
  #    - PUID=1000
  #  restart: on-failure
  #  healthcheck:
  #    test: ["CMD", "curl", "-f", "http://127.0.0.1"]
  #    interval: 30s
  #    timeout: 10s
  #    retries: 5

  dockermon:
    container_name: dockermon
    image: tribunex/ha-dockermon-pi
    ports:
      - 8126:8126
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - /var/docker/config/dockermon:/config
    restart: on-failure

networks:
    gateway:
       external:
          name: gateway
